### ISSUE TITLE
Discuss how to refactor the MenuItem component

### ISSUE THREAD

**Issue by Julesssss** [Expensify/App#30906]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

We would like to refactor the `MenuItem` component. Before making this external we need to agree on what the plan is. 

Suggestions: 

@roryabraham:
> Maybe this isn't the best way to do it, but the way I'd approach this is to look at all the usages we have of MenuItem today to determine the permutations we already have in place, and go from there. 

@lindboe suggested speaking with the @Expensify/design team first: 
> If there's not clear boundaries between what's a separate permutation and what's not, it can be helpful to agree on structured design language so everybody's on the same page about what a "component that looks like this" should be able to do (as a contrived example, "A MenuItem with Badge Text should never also show an RBR indicator")
If there's any forward-looking thinking about MenuItem types (i.e., "We always intended to...") that can also be useful to know

> My personal thoughts: from just the stories, I'm not seeing too many unique cases of content, but instead things that do act like flags (i.e., RBR indicator, focused, disabled states). But I know the stories might not be exhaustive so we'll have to go looking. Even so, even if we don't find clear permutations of content to break out, I think we can still decompose MenuItem into smaller parts.

- [Internal slack discussion](https://expensify.slack.com/archives/C05EQBQG6E5/p1699041701481029?thread_ts=1699017414.494079&cid=C05EQBQG6E5)

cc @roryabraham @lindboe @cdanwards

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0175d7d07fed937e12</li>
        <li>Upwork Job ID: 1721495493814456320</li>
        <li>Last Price Increase: 2023-11-06</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @cdanwards</details>

**Comment by roryabraham** [comment]
I think this is a good idea but don't have a ton of bandwidth to prioritize it myself unfortunately. I will sign on to supervise this but it's not the highest priority item I have on my plate.

**Comment by Julesssss** [comment]
I'll happily handle this!

**Comment by cdanwards** [comment]
Commenting for assignment

**Comment by Julesssss** [comment]
@roryabraham are you happy to handle this? You mentioned being busy, so let me know if you want me to take over. Else I'll unsubscribe and focus on other things.

**Comment by roryabraham** [comment]
@Julesssss feel free to take this from me if you want it. Otherwise I'm happy to keep it too

**Comment by Julesssss** [comment]
I'll leave it assigned to you for now, I offered because this sounded like you didn't have time

> I think this is a good idea but don't have a ton of bandwidth to prioritize it myself unfortunately.

**Comment by cdanwards** [comment]
@Julesssss @roryabraham we're circling around to start some work on this. Is this process similar to level of detail needed for a design doc? Or is it more of "here's what we're thinking" and the discussion/collaboration happens?

**Comment by Julesssss** [comment]
I'd say the latter. We definately don't need a design doc for this, but maybe once we have a plan we can share in Slack for feedback

**Comment by roryabraham** [comment]
@cdanwards any update here? Looking forward to the slack discussion. Let me know if you have any questions or need support.

**Comment by lindboe** [comment]
@Julesssss you closed the other refactor ticket since Attendee Tracking is on hold. Are you thinking about doing the same here, or is it still worth it to work on this independently?

**Comment by Julesssss** [comment]
Hey @lindboe, I think we can go ahead immediately. Would you or @cdanwards be interested in working on this with me?

**Comment by lindboe** [comment]
I'll chat with the team. My understanding is that the Violations implementation work should come first, and I'm trying to get Native Share ready to go to implementation at the moment as well, since that's now also high priority, so it might be a little bit before we can re-focus on this.

**Comment by Julesssss** [comment]
Closing as low priority
